group-actors/src/main/java/org/sunbird/actors/ReadGroupActor.java

```diff
@@ -35,15 +39,36 @@ public void onReceive(Request request) throws Throwable {
    *
    * @param actorMessage
    */
-  private void readGroup(Request actorMessage) throws BaseException {
+  private void readGroup(Request actorMessage) throws Exception {
     logger.info("ReadGroup method call");
     String groupId = (String) actorMessage.getRequest().get(JsonKey.GROUP_ID);
     List<String> requestFields = (List<String>) actorMessage.getRequest().get(JsonKey.FIELDS);
-
+    GroupResponse groupResponse;
+    String groupInfo = cacheUtil.getCache(groupId);
+    if (StringUtils.isNotEmpty(groupInfo)) {
+      groupResponse = JsonUtils.deserialize(groupInfo, GroupResponse.class);
+    } else {
+      groupResponse = groupService.readGroup(groupId);
+      cacheUtil.setCache(groupId, JsonUtils.serialize(groupResponse), self());
+    }
+    if (requestFields.contains(JsonKey.MEMBERS)) {
+      String groupMember = cacheUtil.getCache(groupId + "_" + JsonKey.MEMBERS);
```

Comment:
This concatenation of groupId _ members literal must itself be encapsulated in a function. Example: constructRedisId(groupId)

Reply:
done